src/components/PurchaseOrder/PODetails/PODetailsForm.js

```diff
@@ -14,12 +14,17 @@ class PODetailsForm extends Component {
     change: PropTypes.func,
   }
 
-  static getDerivedStateFromProps(props, state) {
+  static getDerivedStateFromProps(props, { assignToName, createdByName, vendorName }) {
     const { dispatch, change } = props;
-    if (props.vendorName !== state.vendorName || props.assignToName !== state.assignToName) {
+    const isDataChanged = props.vendorName !== vendorName || props.assignToName !== assignToName || props.createdByName !== createdByName;
```

Comment:
I was wondering if there could be any ways to prevent writing a lot of conditions?

Reply:
there is might be a way to avoid this, but I prefer to leave it as is for now. It was working before me and I think it's early to provide such a big refactoring now. I suggest to make it later, when I know common architecture stripes connect patterns better